worker/remoterelations/remoterelations_test.go

```diff
@@ -0,0 +1,241 @@
+	// worker is killed.
+	w := s.assertRemoteRelationsWorkers(c)
+	defer workertest.CleanKill(c, w)
+	s.stub.ResetCalls()
+
+	unitsWatcher, _ := s.facade.updateRelationLife("db2:db django:db", params.Dead)
+	relWatcher, _ := s.facade.remoteApplicationRelationsWatcher("django")
+	relWatcher.changes <- []string{"db2:db django:db"}
+	for a := coretesting.LongAttempt.Start(); a.Next(); {
+		_, ok := s.facade.relationsUnitsWatcher("db2:db django:db")
+		if !ok {
+			break
+		}
+	}
+	c.Assert(unitsWatcher.killed(), jc.IsTrue)
+	expected := []jujutesting.StubCall{
+		{"RemoteRelations", []interface{}{[]string{"db2:db django:db"}}},
+	}
+	s.waitForStubCalls(c, expected)
+}
+
+func (s *remoteRelationsSuite) TestRemoteRelationsRemoved(c *gc.C) {
+	// Checks that when a remote relation goes away, the relation units
+	// worker is killed.
+	w := s.assertRemoteRelationsWorkers(c)
+	defer workertest.CleanKill(c, w)
+	s.stub.ResetCalls()
+
+	unitsWatcher, _ := s.facade.removeRelation("db2:db django:db")
+	relWatcher, _ := s.facade.remoteApplicationRelationsWatcher("django")
+	relWatcher.changes <- []string{"db2:db django:db"}
+	for a := coretesting.LongAttempt.Start(); a.Next(); {
+		_, ok := s.facade.relationsUnitsWatcher("db2:db django:db")
+		if !ok {
+			break
+		}
+	}
+	c.Assert(unitsWatcher.killed(), jc.IsTrue)
+	expected := []jujutesting.StubCall{
+		{"RemoteRelations", []interface{}{[]string{"db2:db django:db"}}},
+	}
+	s.waitForStubCalls(c, expected)
+}
+
+func (s *remoteRelationsSuite) TestRemoteRelationsChangedNotifies(c *gc.C) {
+	var gotChange watcher.RelationUnitsChange
+	changed := make(chan bool)
+	s.PatchValue(&remoterelations.ObserverRelationUnitsChange,
+		func(change watcher.RelationUnitsChange) error {
+			gotChange = change
+			changed <- true
+			return nil
+		})
+	w := s.assertRemoteRelationsWorkers(c)
+	defer workertest.CleanKill(c, w)
+	s.stub.ResetCalls()
+
+	unitsWatcher, _ := s.facade.relationsUnitsWatcher("db2:db django:db")
+	unitsWatcher.changes <- watcher.RelationUnitsChange{
+		Changed:  map[string]watcher.UnitSettings{"unit1": {Version: 2}},
+		Departed: []string{"unit2"},
+	}
+	select {
+	case <-changed:
+	case <-time.After(coretesting.LongWait):
+		c.Fatal("waiting for relation changes")
+	}
+	c.Assert(gotChange, jc.DeepEquals, watcher.RelationUnitsChange{
+		Changed:  map[string]watcher.UnitSettings{"unit1": {Version: 2}},
+		Departed: []string{"unit2"},
+	})
+}
+
+func (s *remoteRelationsSuite) TestRemoteRelationsChangedError(c *gc.C) {
+	s.PatchValue(&remoterelations.ObserverRelationUnitsChange,
+		func(change watcher.RelationUnitsChange) error {
+			return errors.New("failed")
+		})
+	w := s.assertRemoteRelationsWorkers(c)
+
+	unitsWatcher, _ := s.facade.relationsUnitsWatcher("db2:db django:db")
+	unitsWatcher.changes <- watcher.RelationUnitsChange{
+		Changed:  map[string]watcher.UnitSettings{"unit1": {Version: 2}},
+		Departed: []string{"unit2"},
+	}
+	err := workertest.CheckKill(c, w)
```

Comment:
This should be CheckKill*ed*. The worker should stop on account of the RUW failing; you shouldn't need to prompt it.